### ISSUE TITLE
Cannot add Beca BAC-002 devices

### ISSUE THREAD

**Issue by luigixx73** [make-all/tuya-local#950]
I had 6 thermostats that were successfully recognized by the version 2023.6.1 as beca_bac_002.
After updating to the last version 2023.7 I was unable to add a new thermostat of the same model.
I deleted the previous working thermostats to check if there was something wrong with this thermostat and I weren't able to add them again.
All the thermostats work well using Tuya app and the device ID and local  keys weren't changed.
I tried to revert back the update trying to download the previous 2023.6.1 version from HACS but I doesn't work. I think that the previous repositories have been deleted.
Please check this issue with the new version or at least give us the possibility to come back to 2323.6.1 as this version was working nicely.
Thank you.

**Comment by make-all** [comment]
With no logs, there is nothing for me to check.

**Comment by luigixx73** [comment]
Sorry, I was able to downgrade to 2023.6.1.
The devices are detected asfweit_dimmer in both 2023.6.1 and 2023.7 
Very weird as they were added well days ago with 2023.6.1

This is the log:
Logger: custom_components.tuya_local.config_flow
Source: custom_components/tuya_local/config_flow.py:49
Integration: Tuya Local ([documentation](https://github.com/make-all/tuya-local), [issues](https://github.com/make-all/tuya-local/issues))
First occurred: 00:06:13 (2 occurrences)
Last logged: 00:06:13

Device matches feit_dimmer with quality of 43%. DPS: {"updated_at": 1690754770.6754258, "1": false, "2": 49, "3": 53, "5": false, "6": false, "102": "0", "103": "0"}
Report this to https://github.com/make-all/tuya-local/issues/

**Comment by make-all** [comment]
There are some disruptive changes in main, so I cannot recommend upgrading to that right now, but the devices seem to be missing the schedule dp.  It may come back if they are turned on when you add them, or it may be due to a firmware update changing that to a write-only dp.  If you manually add the `optional: true` marking to the schedule in the config file, it should resolve the problem.

**Comment by luigixx73** [comment]
I added the line as you indicate and get this:
Error handling request
00:36:08 – (ERROR) util/yaml/loader.py
Unable to read file /config/custom_components/tuya_local/devices/._beca_bac002_thermostat_c.yaml: 'utf-8' codec can't decode byte 0xb0 in position 37: invalid start byte
00:36:08 – (ERROR) util/yaml/loader.py

**Comment by luigixx73** [comment]
Great! It works.... however you must edit the file within docker if you HA is docker based install, don't open the device file with VSC/Samba as there are some issues with the text encoding, probably you need to save it as ascii instead UTF-8.
But the important thing, it works!
Thanks a lot Sir.

**Comment by luigixx73** [comment]
Yes, if you open it with VSC you must save the file as ASCII encoded, it opens and convert it to by default as UTF-8 so it produces the issues from above.

**Comment by make-all** [comment]
The contents are the same in ASCII and UTF-8, as there are no non-ASCII characters in the original file.  The Latin-1 character at B0 happens to be the degree symbol, which does not appear in the file.

**Comment by make-all** [comment]
It might be useful to add the product id to the matching config for future reference, but it isn't clear to me whether the match is for BAC-002 (the subject of this issue), or Moes BHT-002 since you mentioned a Moes model having the same product id?

**Comment by xbmcnut** [comment]
![image](https://github.com/make-all/tuya-local/assets/1492137/a7a4bef6-8ac5-4834-a930-6513d2ffc742)

Here are the two I have. The Moes added without issue. The 'clone' with the colour screen was not recognised as a thermostat until 2023.8.x. Let me know if you need more info from the dev console.

**Comment by make-all** [comment]
To clarify, it might help if you download the diagnostics for the 16A-3000BH device from Settings / Device & Services / Devices / [16A-3000BH Device] / Download diagnostics

**Comment by xbmcnut** [comment]
Done. https://pastebin.com/BRt6z9Dk

**Comment by xbmcnut** [comment]
> ![image](https://user-images.githubusercontent.com/1492137/260606706-a7a4bef6-8ac5-4834-a930-6513d2ffc742.png)
> 
> Here are the two I have. The Moes added without issue. The 'clone' with the colour screen was not recognised as a thermostat until 2023.8.x. Let me know if you need more info from the dev console.

I went to add the 'Moes' version into production today and oddly, it won't connect at all even though above, I was able to add it seamlessly back in August. I'm now running 5.2.1 under HA 2023.11.1. Of course, I pulled the new key from Tuya cloud.

In the cloud console, this device is showing as read only but my other floor thermostat is showing as 'controllable'.

![image](https://github.com/make-all/tuya-local/assets/1492137/e1144ff4-ff84-4f85-8962-bf55e71983b8)

**Comment by timowua** [comment]
I got the same problem. I have 10 thermostats with same product_id: "IAYz2WK1th0cMLmL". During the adding process, 2 of them were not recognized. I checked the log and found that they are not sending the DP 4 (mode). On iot.tuya.com all devices look the same, all have 4th DP when I check it Query Device Details. But somehow two of them doesn't transfer it to tuya_local.
HAC 2024.2.3

Here are diagnostics confings for normal and faulty devices
[config_entry-tuya_local-864beba7586ff1860444c502eb66a09b.json](https://github.com/make-all/tuya-local/files/14412732/config_entry-tuya_local-864beba7586ff1860444c502eb66a09b.json)
[config_entry-tuya_local-e37c635b61b94a3386f9217e53ff6dbc.json](https://github.com/make-all/tuya-local/files/14412733/config_entry-tuya_local-e37c635b61b94a3386f9217e53ff6dbc.json)
